stingray/deadtime/fad.py

```diff
@@ -0,0 +1,189 @@
+    ptot = 0
+    cs = 0
+    n = 0
+    average_diff = average_diff_uncorr = 0
+
+    if plot:
+        plt.figure()
+
+    for start_ind, end_ind in zip(start_inds, end_inds):
+        freq, f1, f1_leahy = _get_fourier_intv(lc1, start_ind, end_ind)
+        freq, f2, f2_leahy = _get_fourier_intv(lc2, start_ind, end_ind)
+        freq, ftot, ftot_leahy = \
+            _get_fourier_intv(summed_lc, start_ind, end_ind)
+
+        fourier_diff = f1_leahy - f2_leahy
+
+        if smoothing_alg == 'gauss':
+            smooth_real = gaussian_filter1d(fourier_diff.real ** 2,
+                                            smoothing_length)
+        else:
+            raise ValueError("Unknown smoothing algorithm: {}".format(
+                smoothing_alg))
+
+        if plot:
+            plt.scatter(freq, fourier_diff, s=1)
+
+        if all_leahy:
+            f1 = f1_leahy
+            f2 = f2_leahy
+            ftot = ftot_leahy
+        p1 = (f1 * f1.conj()).real
+        p1 = p1 / smooth_real * 2
+        p2 = (f2 * f2.conj()).real
+        p2 = p2 / smooth_real * 2
+        pt = (ftot * ftot.conj()).real
+        pt = pt / smooth_real * 2
+
+        c = (f2 * f1.conj()).real
+        c = c / smooth_real * 2
+
+        if n == 0 and plot:
+            plt.plot(freq, smooth_real, zorder=10, lw=3)
+            plt.plot(freq, f1_leahy, zorder=5, lw=1)
+            plt.plot(freq, f2_leahy, zorder=5, lw=1)
+
+        ptot += pt
+        pds1 += p1
+        pds2 += p2
+        cs += c
+        average_diff += fourier_diff / smooth_real ** 0.5 * np.sqrt(2)
+        average_diff_uncorr += fourier_diff
+        n += 1
+
+    std = (average_diff / n).std()
+    stdtheor = 2 / np.sqrt(n)
+    stduncorr = (average_diff_uncorr / n).std()
+    is_compliant = np.abs((std - stdtheor) / stdtheor) < tolerance
+    verbose_string = \
+        '''
+    -------- FAD correction ----------
+    I smoothed over {smoothing_length} power spectral bins
+    {n} intervals averaged.
+    The uncorrected standard deviation of the Fourier
+    differences is {stduncorr} (dead-time affected!)
+    The final standard deviation of the FAD-corrected
+    Fourier differences is {std}. For the results to be
+    acceptable, this should be close to {stdtheor}
+    to within {tolerance} %.
+    In this case, the results ARE {compl}complying.
+    {additional}
+    ----------------------------------
+    '''.format(smoothing_length=smoothing_length,
+               n=n,
+               stduncorr=stduncorr,
+               std=std,
+               stdtheor=stdtheor,
+               tolerance=tolerance * 100,
+               compl='NOT ' if not is_compliant else '',
+               additional='Maybe something is not right.' if not is_compliant else '')
+
+    print(verbose_string)
+    if verbose and is_compliant:
+        log.info(verbose_string)
+    elif not is_compliant:
+        warnings.warn(verbose_string)
+
+    if strict:
+        assert is_compliant
+
+    results = Table()
```

Comment:
So, I would like to save all results together as a single table, mostly to save space. I introduced a couple of new things:
1. The table with the results can be saved to a file. This might be useful in case we need to dump the results of a large calculation for future use.
2. I wrote a new `get_periodograms_from_FAD_results` function that can accept either the table of the FAD correction results or the output file where the results are saved, and return the corresponding Stingray objects (e.g. `get_periodograms_from_FAD_results(results, kind='ptot')` returns the PDS from the summed light curves, kind='cs' the cospectrum etc.).
Do you think it's a good compromise?

Reply:
I get the space saving issue, but I also think that running basically *any* stingray function and getting back an Astropy Table would have me very confused, because that isn't really what we return in any other part of the code. Is dumping the results to file the most likely use case? Are we very worried about the size of periodograms that we need to avoid duplication? I'm not trying to be difficult, I genuinely would like to understand where the bottlenecks are. 
It might be valuable to have a boolean option on the FAD function `return_objects` that if true, will return the objects, otherwise an Astropy Table, and which can be set to `False` by default if we think that most users will want a table.

Reply:
Ok, `return_objects` implemented and tested :)